include/c_utilities/filesystem.h

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef C_UTILITIES__FILESYSTEM_H_
+#define C_UTILITIES__FILESYSTEM_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+
+#include "c_utilities/macros.h"
+#include "c_utilities/visibility_control.h"
+
+/// Return current working directory.
+/**
+ * \param[in] buffer Allocated string to store current directory path to
+ * \param[in] max_length maximum length to be stored in buffer
+ * \return bool True if success False otherwise
+ */
+C_UTILITIES_PUBLIC
+C_UTILITIES_WARN_UNUSED
+bool
+utilities_get_cwd(char * buffer, size_t max_length);
```

Comment:
good point. For every other function, the check on `stat` will cause a `return false` so I think only this function needs to be modified.

I didn't see the need to go all the way to create error codes for these because the scope of these function is pretty limited. But if you think otherwise I can add some